pkg/scheduler/core/generic_scheduler.go

```diff
@@ -491,6 +491,7 @@ func (g *genericScheduler) findNodesThatFit(pod *v1.Pod, nodes []*v1.Node) ([]*v
 				predicateResultLock.Lock()
 				errs[err.Error()]++
 				predicateResultLock.Unlock()
+				cancel()
```

Comment:
We shouldn't cancel the whole context here. As it's intentional to have all `checkNode()` finished so as to get a final aggregated `[]errs`.

Reply:
There is numNodesToFind beyond which the checkNode() would return early.

Do you think the same threshold can be applied to len(errs) ? When len(errs) > numNodesToFind, we can cancel.

Cheers

Reply:
Nope. `len(errs)` can't be compared with `numNodesToFind`. If you look at the data structure:

```go
errs := errors.MessageCountMap{}
```

`errs` denotes number of errors per ErrorType.

Reply:
len(errs) was simplified measure in my previous comment.
What if we go over the map and add up number of errors ? When the sum is greater than numNodesToFind (or some new threshold), we cancel the context.

Thanks

Reply:
Since the number of nodes to find is bounded in the normal case, it would be intuitive to bound the errors as well.

I don't think the caller is able to get more out of all the errors versus bounded number of errors.

Reply:
Ranging a map has cost. We shouldn't do optimization for the 1% case (even without benchmark number yet) with the cost of hurting the other 99%.